pyccel/codegen/pipeline.py

```diff
@@ -234,13 +242,16 @@ def handle_error(stage):
         handle_error('annotation (semantic)')
         raise PyccelSemanticError('Semantic step failed')
 
+    end_semantic = time.time()
+
     if semantic_only:
         pyccel_stage.pyccel_finished()
         return
```

Comment:
Same comment as above. Other `return` statements are on lines 285, 332, and 343.